src/org/labkey/test/components/ui/grids/ResponsiveGrid.java

```diff
@@ -107,9 +109,78 @@ public T sortColumnDescending(String columnLabel)
         return getThis();
     }
 
-    private void sortColumn(String columnLabel, SortDirection direction)
+    public void sortColumn(String columnLabel, SortDirection direction)
+    {
+        clickColumnMenuItem(columnLabel, direction.equals(SortDirection.DESC) ? "Sort descending" : "Sort ascending", true);
+    }
+
+    public void clearSort(String columnLabel)
+    {
+        clickColumnMenuItem(columnLabel, "Clear sort", true);
+    }
+
+    public boolean hasColumnSortIcon(String columnLabel)
+    {
+        WebElement headerCell = elementCache().getColumnHeaderCell(columnLabel);
+        Optional<WebElement> colHeaderIcon = Locator.XPathLocator.union(
+                Locator.tagWithClass("span", "fa-sort-amount-asc"),
+                Locator.tagWithClass("span", "fa-sort-amount-desc")
+        ).findOptionalElement(headerCell);
+        return colHeaderIcon.isPresent();
+
+    }
+
+    public T filterColumn(String columnLabel, Filter.Operator operator)
+    {
+        return filterColumn(columnLabel, operator, null);
+    }
+
+    public T filterColumn(String columnLabel, Filter.Operator operator, Object value)
+    {
+        T _this = getThis();
+        GridFilterModal filterModal = initFilterColumn(columnLabel, operator, value);
+        filterModal.confirm();
+        return _this;
+    }
+
+    public T filterColumn(String columnLabel, Filter.Operator operator1, Object value1, Filter.Operator operator2, Object value2)
+    {
+        T _this = getThis();
+        GridFilterModal filterModal = initFilterColumn(columnLabel, null, null);
+        filterModal.selectExpressionTab().setFilters(
+                new FilterExpressionPanel.Expression(operator1, value1),
+                new FilterExpressionPanel.Expression(operator2, value2)
+        );
+        filterModal.confirm();
+        return _this;
+    }
+
+    public String filterColumnExpectingError(String columnLabel, Filter.Operator operator, Object value)
+    {
+        GridFilterModal filterModal = initFilterColumn(columnLabel, operator, value);
+        filterModal.confirm(true);
+        String errorMsg = filterModal.getErrorMsg();
+        filterModal.cancel();
+        return errorMsg;
+    }
+
+    private GridFilterModal initFilterColumn(String columnLabel, Filter.Operator operator, Object value)
+    {
+        clickColumnMenuItem(columnLabel, "Filter...", false);
+        GridFilterModal filterModal = new GridFilterModal(getDriver(), getThis()::doAndWaitForUpdate);
```

Comment:
ResponsiveGrid implements `UpdatingComponent` interface that the constructor takes. This shouldn't need to use a method reference.
```suggestion
        GridFilterModal filterModal = new GridFilterModal(getDriver(), this);
```

Reply:
Yup, good catch. Done